README.md

```diff
@@ -1,136 +1,75 @@
 
-## Test Modes
+The library is composed of several modules. The most important of these is the [`pulumitest`](./pulumitest/) module. This is a library designed for testing any Pulumi program within a Go test. It extends the Go [Automation API](https://www.pulumi.com/automation/) with defaults appropriate for local testing such as using temporary directories for state.
 
-### End To End (e2e)
-
-Purpose: Prove provider behaviour for a resource is correct.
-
-This test mode does not use the SDKs, but executes the YAML program directly. There are two types of e2e test: quick and full:
-
-1. Quick will not provision real resources with the cloud provider.
-2. Full will test the whole lifecycle of resources.
-
-### SDK
-
-Purpose: Ensure parity of each supported language's SDK behaviour.
-
-SDK test are therefore split out per-language. Internally, this uses `pulumi convert` to automatically create the language-specific programs before executing them.
-
-### VerifyUpgrade
-
-Purpose: Verifies that upgrading the provider does not generate any unexpected replacements.
-
-What these tests specifically try to verify is that the provider binary release candidate under test
-will not generate any surprises for users attempting to upgrade to it from the baseline released
-version.
-
-There are currently several
-[UpgradeTestMode](https://github.com/search?q=repo%3Apulumi%2Fprovidertest+type+UpgradeTestMode&type=code)
-variations tests can run under, with different speed/accuracy trade-offs.
-
-
-## Example Usage
+Here's a short example of how to use pulumitest:
 
 ```go
-func TestSimple(t *testing.T) {
-  test := NewProviderTest("test/simple", // Point to a directory containing a Pulumi YAML program
-    WithProvider(StartLocalProvider), // Provider can be started and attached in-process
-    WithUpdateStep(UpdateStepDir("../simple-2"))) // Multi-step tests are supported
-  test.Run(t)
+import (
+  "filepath"
+  "github.com/pulumi/providertest/pulumitest"
+)
+
+func TestPulumiProgram(t *testing.T) {
+  test := NewPulumiTest(t, filepath.Join("path", "to", "program"))
+  test.Preview(t)
+  test.Up(t)
+  test.Refresh(t)
 }
 ```
 
-When calling `.Run()`, a suite of nested tests are run:
+Refer to [the full documentation](./pulumitest/README.md) for a complete walkthrough of the features.
 
-- `TestSimple/e2e`
-- `TestSimple/sdk-csharp`
-- `TestSimple/sdk-go`
-- `TestSimple/sdk-python`
-- `TestSimple/sdk-typescript`
+## Upgrade Testing
 
-If you want to run just one of these test modes directly locally, then you can temporarily replace `.Run(t)` with:
+We perform "upgrade testing" on providers to indicate when there's a change in a new version of a provider which might cause resources to be re-created during an upgrade from a previous version.
```

Comment:
I had to read this sentence a few times. We don't use upgrade testing so much to indicate new versions but to ensure there are no unexpected resource replacements.